arch/risc-v/src/common/riscv_macros.S

```diff
@@ -58,7 +58,9 @@
 #ifdef RISCV_SAVE_GP
   REGSTORE   x3,  REG_X3(\in)    /* gp */
 #endif
+#ifdef CONFIG_SCHED_THREAD_LOCAL
```

Comment:
Unnecessary as well, TP will remain stable when the user process is running, as well as during a trap. Only time when TP needs to be updated is when a context switch occurs.